state/executor.go

```diff
@@ -355,6 +355,16 @@ func NewTransitionApplicationError(err error, isRecoverable bool) *TransitionApp
 	}
 }
 
+type GasLimitReachedTransitionApplicationError struct {
+	TransitionApplicationError
+}
```

Comment:
Is there any specific reason why we wrap this error type?

Reply:
I think it's because he's trying pass that specific error up [WriteTransactions()](https://github.com/0xPolygon/polygon-sdk/blob/121570edc4ebc88532e14459886fc14f96b98d4e/consensus/ibft/ibft.go#L443) (previously in `buildBlock()`). 

I guess that's a bit problematic to achieve due to the way we cast these errors up the call stack.

Reply:
I didn't want to add additional field in the TransitionApplicationError so I created this one. The problem was to determine when it is the time to break from the loop, and the moment we chose was when the next transaction can't fit in the block due to the block gas limit being reached.